textattack/shared/attack.py

```diff
@@ -86,15 +105,31 @@ def __init__(
         )
         self.search_method.filter_transformations = self.filter_transformations
 
-    def get_transformations(self, current_text, original_text=None, **kwargs):
+    def _get_transformations_uncached(self, current_text, original_text=None, **kwargs):
         """Applies ``self.transformation`` to ``text``, then filters the list
         of possible transformations through the applicable constraints.
 
         Args:
             current_text: The current ``AttackedText`` on which to perform the transformations.
             original_text: The original ``AttackedText`` from which the attack started.
-            apply_constraints: Whether or not to apply post-transformation constraints.
+        Returns:
+            A filtered list of transformations where each transformation matches the constraints
+        """
+        transformed_texts = self.transformation(
```

Comment:
I removed it because I don't think there's really anything to benefit from using np.array (but correct me if I'm wrong). Numpy arrays are fast for matrix and math-y operations but we have a list of AttackedText, which is an object with varying size. I really don't think creating an numpy array and copying the data again to fit into a contiguous memory block is that beneficial, especially when we'll be filtering stuff again using constraints.